src/main/java/strategy/FruitShopStrategyImpl.java

```diff
@@ -0,0 +1,26 @@
+package strategy;
+
+import db.ShopStorage;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+
+public class FruitShopStrategyImpl {
+    private final Map<FruitTransaction.Operation, OperationHandler> fruitShopStrategy;
+
+    public FruitShopStrategyImpl() {
+        fruitShopStrategy = new HashMap<>();
+        fruitShopStrategy.put(FruitTransaction.Operation.BALANCE, new BalanceHandler());
+        fruitShopStrategy.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler());
+        fruitShopStrategy.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler());
+        fruitShopStrategy.put(FruitTransaction.Operation.RETURN, new ReturnHandler());
+    }
+
+    public void processTransactions(List<FruitTransaction> transactions, ShopStorage fruitStorage) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = fruitShopStrategy.get(transaction.getOperation());
+            handler.handle(transaction, fruitStorage);
+        }
+    }
```

Comment:
Passing services we rely on as method arguments is not a good decision, please use them as constructor arguments:
```suggestion
    /* This part is extracted to Main class: */
        
        Map<FruitTransaction.Operation, OperationHandler> fruitShopStrategyMap = new HashMap<>();
        fruitShopStrategy.put(FruitTransaction.Operation.BALANCE, new BalanceHandler(fruitStorage));
        fruitShopStrategy.put(FruitTransaction.Operation.SUPPLY, new SupplyHandler(fruitStorage));
        fruitShopStrategy.put(FruitTransaction.Operation.PURCHASE, new PurchaseHandler(fruitStorage));
        fruitShopStrategy.put(FruitTransaction.Operation.RETURN, new ReturnHandler(fruitStorage));
     
     /**/


    public FruitShopStrategyImpl(Map<FruitTransaction.Operation, OperationHandler> fruitShopStrategyMap) {
        this. fruitShopStrategyMap = fruitShopStrategyMap;
    }

    public void processTransactions(List<FruitTransaction> transactions) {
    for (FruitTransaction transaction : transactions) {
           fruitShopStrategy.get(transaction.getOperation()).handle(transaction);
    }
```